src/Integration/Binding/DotNetBindingConfigProvider.cs

```diff
@@ -44,83 +45,98 @@ internal class DotNetBindingConfigProvider : IBindingConfigProvider
         private readonly string serverUrl;
         private readonly string projectName;
 
+        private readonly IRuleSetGenerator ruleSetGenerator;
+        private readonly INuGetPackageInfoGenerator nuGetPackageInfoGenerator;
+
         public DotNetBindingConfigProvider(ISonarQubeService sonarQubeService, INuGetBindingOperation nuGetBindingOperation, string serverUrl, string projectName, ILogger logger)
+            : this(sonarQubeService, nuGetBindingOperation, serverUrl, projectName, logger,
+                  new RuleSetGenerator(), new NuGetPackageInfoGenerator())
+        {
+        }
+
+        internal /* for testing */ DotNetBindingConfigProvider(ISonarQubeService sonarQubeService, INuGetBindingOperation nuGetBindingOperation, string serverUrl, string projectName, ILogger logger,
+            IRuleSetGenerator ruleSetGenerator, INuGetPackageInfoGenerator nuGetPackageInfoGenerator)
         {
             this.sonarQubeService = sonarQubeService;
             this.nuGetBindingOperation = nuGetBindingOperation;
             this.serverUrl = serverUrl;
             this.projectName = projectName;
             this.logger = logger;
+
+            this.ruleSetGenerator = ruleSetGenerator;
+            this.nuGetPackageInfoGenerator = nuGetPackageInfoGenerator;
         }
 
         public bool IsLanguageSupported(Language language)
         {
             return Language.CSharp.Equals(language) || Language.VBNET.Equals(language);
         }
 
-        public async Task<IBindingConfigFile> GetConfigurationAsync(SonarQubeQualityProfile qualityProfile, string organizationKey, Language language, CancellationToken cancellationToken)
+        public Task<IBindingConfigFile> GetConfigurationAsync(SonarQubeQualityProfile qualityProfile, string UNUSED_organizationKey_TODO_REMOVE, Language language, CancellationToken cancellationToken)
```

Comment:
I've split this method into two parts as recommended by S4457 - arg checks in the non-async part, async part in a separate method.